lib/modules/manager/bazel-module/rules.ts

```diff
@@ -0,0 +1,147 @@
+import is from '@sindresorhus/is';
+import parseGithubUrl from 'github-url-from-git';
+import { z } from 'zod';
+import { logger } from '../../../logger';
+import type { SkipReason } from '../../../types';
+import { regEx } from '../../../util/regex';
+import { BazelDatasource } from '../../datasource/bazel';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import type { PackageDependency } from '../types';
+import { RecordFragmentSchema, StringFragmentSchema } from './fragments';
+
+// Rule Schemas
+
+export interface BasePackageDep extends PackageDependency {
+  depType: string;
+  depName: string;
+}
+
+export interface OverridePackageDep extends BasePackageDep {
+  // This value is set as the skipReason on the bazel_dep PackageDependency.
+  bazelDepSkipReason: SkipReason;
+}
+
+type BazelModulePackageDep = BasePackageDep | OverridePackageDep;
+
+function isOverride(value: BazelModulePackageDep): value is OverridePackageDep {
+  return 'bazelDepSkipReason' in value;
+}
+
+// This function exists to remove properties that are specific to
+// OverridePackageDep. In theory, there is no harm in leaving the properties
+// as it does not invalidate the PackageDependency interface. However, it might
+// be surprising to someone outside the bazel-module code to see the extra
+// properties.
+export function overrideToPackageDependency(
+  override: OverridePackageDep
+): PackageDependency {
+  const copy: Partial<OverridePackageDep> = structuredClone(override);
+  delete copy.bazelDepSkipReason;
+  return copy;
+}
+
+const BazelDepToPackageDep = RecordFragmentSchema.extend({
+  children: z.object({
+    rule: StringFragmentSchema.extend({
+      value: z.literal('bazel_dep'),
+    }),
+    name: StringFragmentSchema,
+    version: StringFragmentSchema,
+  }),
+}).transform(
+  ({ children: { rule, name, version } }): BasePackageDep => ({
+    datasource: BazelDatasource.id,
+    depType: rule.value,
+    depName: name.value,
+    currentValue: version.value,
+  })
+);
+
+const GitOverrideToPackageDep = RecordFragmentSchema.extend({
+  children: z.object({
+    rule: StringFragmentSchema.extend({
+      value: z.literal('git_override'),
+    }),
+    module_name: StringFragmentSchema,
+    remote: StringFragmentSchema,
+    commit: StringFragmentSchema,
+  }),
+}).transform(
+  ({
+    children: { rule, module_name: moduleName, remote, commit },
+  }): OverridePackageDep => {
+    const override: OverridePackageDep = {
+      depType: rule.value,
+      depName: moduleName.value,
+      bazelDepSkipReason: 'git-dependency',
+      currentDigest: commit.value,
+    };
+    if (isGithubRemote(remote.value)) {
```

Comment:
I would prefer to 
```suggestion
    const packageName = githubPackageName(remote.value);
    if (packageName) {
```

so we don't need to parse url twice (perf)